app/src/main/java/org/liberty/android/fantastischmemo/utils/RecentListActionModeUtil.java

```diff
@@ -0,0 +1,123 @@
+package org.liberty.android.fantastischmemo.utils;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.view.ActionMode;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import org.liberty.android.fantastischmemo.R;
+import org.liberty.android.fantastischmemo.common.BaseActivity;
+import org.liberty.android.fantastischmemo.modules.PerActivity;
+
+import javax.inject.Inject;
+
+@PerActivity
+public class RecentListActionModeUtil {
+
+    private final BaseActivity mContext;
+
+    private final ActionModeCallback mActionModeCallback = new ActionModeCallback();
+
+    private ActionMode mActionMode;
+
+    private BroadcastReceiver mStartActionMode;
+
+    private BroadcastReceiver mUpdateActionMode;
+
+    private BroadcastReceiver mStopActionMode;
+
+    @Inject
+    public RecentListActionModeUtil(BaseActivity activityContext) {
+        this.mContext = activityContext;
+    }
+
+
+    public void register() {
+        mStartActionMode = new BroadcastReceiver() {
```

Comment:
I'm not sure if I understand it right. I would still have to have a register() and deregister() method for registering and unregistering, right?